core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -480,26 +502,42 @@ class CityConstructions {
      *  @param automatic        Flag whether automation should try to choose what next to build (not coming from UI)
      *                          Note: settings.autoAssignCityProduction is handled later
      *  @param stat             Stat object of the stat with which was paid for the construction
-     *  @return                 Success (false e.g. unit cannot be placed
+     *  @param tile             Supports [UniqueType.CreatesOneImprovement] the tile to place the improvement from that unique on.
+     *                          Ignored when the [constructionName] does not have that unique. If null and the building has the unique, a tile is chosen automatically.
+     *  @return                 Success (false e.g. unit cannot be placed)
      */
     fun purchaseConstruction(
-        constructionName: String, 
-        queuePosition: Int, 
-        automatic: Boolean, 
-        stat: Stat = Stat.Gold
+        constructionName: String,
+        queuePosition: Int,
+        automatic: Boolean,
+        stat: Stat = Stat.Gold,
+        tile: TileInfo? = null
     ): Boolean {
-        if (!(getConstruction(constructionName) as INonPerpetualConstruction).postBuildEvent(this, stat))
+        val construction = getConstruction(constructionName) as? INonPerpetualConstruction ?: return false
+
+        // Support UniqueType.CreatesOneImprovement: it is active when getImprovementToCreate returns an improvement
+        val improvementToPlace = (construction as? Building)?.getImprovementToCreate(cityInfo.getRuleset())
+        if (improvementToPlace != null) {
+            // If active without a predetermined tile to place the improvement on, automate a tile
+            val finalTile = tile ?: Automation.getTileForConstructionImprovement(cityInfo, improvementToPlace)
+            if (finalTile == null) {
+                println("${cityInfo.name} tried to buy $constructionName without having any suitable tile for a ${improvementToPlace.name}.")  // todo remove when debugged
```

Comment:
:upside_down_face:

Reply:
Ah yes a breakpoint anchor... Do we not have something like `asm { emit(0xCC) }` (if memory serves...) or `ILGenerator.Emit(OpCodes.Break);` ??

That breakpoint never hit.